sktime/base.py

```diff
@@ -4,7 +4,9 @@
 from sklearn.base import _pprint as pprint_sklearn
 
 
-class _base_methods_mixin(object):
+class _base_methods_mixin(object, metaclass=abc.ABCMeta):
+    """ defines common methods used by both Estimator and Model classes.
```

Comment:
this doesn't seem right, do we have models with `fit`?

Reply:
good spot, i moved the getattribute override to Estimator.